designs/multiple-worker-node-groups.md

```diff
@@ -0,0 +1,463 @@
+# Multiple worker node groups
+
+## Introduction
+
+**Problem:** Currently users of eks-anywhere can only create one worker node group by specifying the configuration details in a cluster spec.
+
+This limits the ability of a user to create worker nodes with different configurations and formulate application deployment strategy based on the configurations of the worker nodes.
+
+Also, this creates a problem while tainting a node. Unless, eks-anywhere clusters support multiple worker node groups, worker nodes can not be tainted with `NoExecute` effect, since nodes in a node group have the same configuration, setting a taint with `NoExecute` effect would essentially make the worker nodes unusable for eks-anywhere specific deployments, as the general approach for eks-anywhere is to not add tolerations on deployments.
```

Comment:
I totally agree with this
However, the user could still add `NoExecute` taints to all worker nodes. Which will be a problem, specially in management clusters, since CAPI and EKS-A controllers won't be able to run.

I think it's worth putting in this design what do we plan to do about it.
Imo there are multiple valid solutions: adding validations to prevent this, adding taints to controllers (I don't love this one), separate worker for controllers or even maybe just documenting this and instructing the user not to do it.